src/templates/filtersDrawer.jsx

```diff
@@ -0,0 +1,674 @@
+
+        onItemChange(filteredOptions);
+    }
+
+    static onNestedTogglesValueLabelClick(nestedTogglesData, selectedOption) {
+        const selectedOptions = _.filter(nestedTogglesData.value, (d) => (
+            d.value !== selectedOption.value
+        ));
+
+        nestedTogglesData.onChange(selectedOptions);
+    }
+
+    constructor() {
+        super();
+
+        /**
+         * The nestedTogglesData object is only to be populated when props.rows.items.nestedToggles
+         * is true and a label onClick event handler is triggered.
+         */
+        this.state = {
+            nestedTogglesData: {},
+        };
+
+        this.onApplyClick = this.onApplyClick.bind(this);
+        this.onClearClick = this.onClearClick.bind(this);
+        this.onNestedTogglesCloseWingClick = this.onNestedTogglesCloseWingClick.bind(this);
+        this.onNestedTogglesLabelClick = this.onNestedTogglesLabelClick.bind(this);
+        this.onNestedTogglesWingOptionLabelClick =
+            this.onNestedTogglesWingOptionLabelClick.bind(this);
+    }
+
+    onApplyClick() {
+        const { onApply } = this.props;
+
+        onApply();
+    }
+
+    onClearClick() {
+        const { onClear } = this.props;
+
+        onClear();
+    }
+
+    onNestedTogglesCloseWingClick() {
+        const { nestedTogglesData } = this.state;
+
+        this.setState({
+            nestedTogglesData: {},
+        }, () => {
+            nestedTogglesData.onChange(nestedTogglesData.value);
+        });
+    }
+
+    onNestedTogglesLabelClick(nestedTogglesData) {
+        this.setState({
+            nestedTogglesData: _.cloneDeep(nestedTogglesData),
+        });
+    }
+
+    onNestedTogglesWingOptionLabelClick(selectedOption) {
+        const { nestedTogglesData } = this.state;
+        let selectedOptions;
+
+        if (_.some(nestedTogglesData.value, selectedOption)) { // Subtract
+            selectedOptions = _.filter(nestedTogglesData.value, (d) => (
+                d.value !== selectedOption.value
+            ));
+        } else { // Add
+            selectedOptions = _.sortBy([...nestedTogglesData.value, selectedOption], ['value']);
+        }
+
+        const newNestedTogglesData = _.mapValues(nestedTogglesData, (data, key) => {
+            let newData = data;
+
+            if (key === 'value') {
+                newData = selectedOptions;
+            }
+
+            return newData;
+        });
+
+        this.setState({
+            nestedTogglesData: _.cloneDeep(newNestedTogglesData),
+        });
+    }
+
+    render() {
+        const {
+            children,
+            className,
+            id,
+            isDirty,
+            isFiltering,
+            isOpen,
+            onClose,
+            rows,
+            style,
+        } = this.props;
+        const { nestedTogglesData } = this.state;
+        const containerClasses = ClassNames('ui', 'page--filters_drawer', className);
```

Comment:
Do we need our now classic parameterization of `page` versus `drawer` here?

Reply:
Ding Dang! Yep we do.

Reply:
Fixed.